zerver/tests/test_typed_endpoint.py

```diff
@@ -0,0 +1,596 @@
+        ):
+            call_endpoint(path_only, HostRequestMock(post_data={"path_var": 15, "other": 1}))
+
+        # path_var in the request body is ignored
+        response = call_endpoint(
+            path_only, HostRequestMock(post_data={"path_var": 15, "other": 1}), path_var=10
+        )
+        self.assert_json_success(response, ignored_parameters=["path_var"])
+        self.assertEqual(orjson.loads(response.content)["val"], 11)
+
+        def path_only_default(
+            request: HttpRequest,
+            *,
+            path_var_default: PathOnly[str] = "test",
+        ) -> None:
+            ...
+
+        with self.assertRaisesMessage(
+            AssertionError, "Path-only parameter path_var_default should not have a default value"
+        ):
+            typed_endpoint(path_only_default)
+
+    def test_documentation_status(self) -> None:
+        def documentation(
+            request: HttpRequest,
+            *,
+            foo: Annotated[
+                str,
+                ApiParamConfig(documentation_status=DocumentationStatus.INTENTIONALLY_UNDOCUMENTED),
+            ],
+            bar: Annotated[
+                str, ApiParamConfig(documentation_status=DocumentationStatus.DOCUMENTATION_PENDING)
+            ],
+            baz: Annotated[
+                str, ApiParamConfig(documentation_status=DocumentationStatus.DOCUMENTED)
+            ],
+            paz: PathOnly[int],
+            other: str,
+        ) -> None:
+            ...
+
+        from zerver.lib.request import arguments_map
+
+        view_func_full_name = f"{documentation.__module__}.{documentation.__name__}"
+        typed_endpoint(documentation)
+        # Path-only and non DOCUMENTED parameters should not be added
+        self.assertEqual(arguments_map[view_func_full_name], ["baz", "other"])
+
+    def test_annotated(self) -> None:
+        @typed_endpoint
+        def view1(
+            request: HttpRequest,
+            *,
+            foo: Annotated[
+                Json[int],
+                ApiParamConfig(path_only=True),
+            ],
+        ) -> None:
+            ...
+
+        def view2(
+            request: HttpRequest,
+            user_profile: UserProfile,
+            *,
+            foo: Annotated[Json[int], ApiParamConfig(), ApiParamConfig()],
+        ) -> None:
+            ...
+
+        with self.assertRaisesMessage(
+            AssertionError, "ApiParamConfig can only be defined once per parameter"
+        ):
+            typed_endpoint(view2)
+
+        @typed_endpoint
+        def view3(
+            request: HttpRequest,
+            user_profile: UserProfile,
+            *,
+            foo: Annotated[Json[bool], str, "unrelated"],
+        ) -> bool:
+            return foo
+
+        hamlet = self.example_user("hamlet")
+        result = call_endpoint(
+            view3, HostRequestMock({"foo": orjson.dumps(False).decode()}), hamlet
+        )
+        self.assertFalse(result)
```

Comment:
It might be worth a comment explaining what's supposed to be going on with `view3`; or maybe better would be to rename it if that can clarify usefully.

Might be worth a pass generally; I find `def annotated_with_extra_unrelated_annotations` or whatever to be nicer than comments for making tests like this readable.